ql/math/randomnumbers/xoshiro256starstaruniformrng.cpp

```diff
@@ -0,0 +1,57 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2023 Ralf Konrad Eckel
+
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include <ql/math/randomnumbers/seedgenerator.hpp>
+#include <ql/math/randomnumbers/xoshiro256starstaruniformrng.hpp>
+
+namespace QuantLib {
+
+    Xoshiro256StarStarUniformRng::Xoshiro256StarStarUniformRng(unsigned long long seed)
+    : Xoshiro256StarStarUniformRng(seed, seed, seed, seed) {}
```

Comment:
We might adopt this one: https://prng.di.unimi.it/splitmix64.c

Reply:
That'd be great.

Reply:
Okay. I will add it via `uint64_t SeedGenerator::getInt64() const`.

Reply:
Hmm, SeedGenerator can't be seeded, it always uses the current time for initialization, and it's a singleton so you can't initialize it twice.  I'd leave it alone.

Reply:
Ah, probably I meant something different with using `uint64_t SeedGenerator::getInt64()`:
```c++
    void Xoshiro256StarStarUniformRng::seedInitialization() const {
        if (s0_ == 0 && s1_ == 0 && s2_ == 0 && s3_ == 0) {
            do {
                s0_ = SeedGenerator::instance().getInt64();
                s1_ = SeedGenerator::instance().getInt64();
                s2_ = SeedGenerator::instance().getInt64();
                s3_ = SeedGenerator::instance().getInt64();
            } while (s0_ == 0 || s1_ == 0 || s2_ == 0 || s3_ == 0);
        }
    }
```

Did you mean something like
```c++
/*! If the given seed is 0, a random seed will be chosen based on clock().
 */
Xoshiro256StarStarUniformRng(uint64_t seed) {
  do {
    s0_ = seed != 0 ? seed : SeedGenerator::instance().get();
    Splitmix64 splitmix64(s0_);
    s1_ = splitmix64.next();
    s2_ = splitmix64.next();
    s3_ = splitmix64.next();
  } while (s0_ == 0 && s1_ == 0 && s2_ == 0 && s3_ == 0);
}

/* ! Be sure to choose s0, s1, s2, s3 randomly.
 * Otherwise the results of the first random numbers might not be well distributed.
 * Especially s0 = s1 = s2 = s3 = 0 does not work and will always return 0.
 */
Xoshiro256StarStarUniformRng(uint64_t s0, uint64_t s1, uint64_t s2, uint64_t s3)
 : s0_(s0), s1_(s1), s2_(s2), s3_(s3) {}
```
which would avoid burning the first 1,000 `nextInt64();`?

Looks to me much cleaner?!

Reply:
Something like that, yes, but as I understand the comment in the original code the initialization in the single-seed case would be more like
```
    Splitmix64 splitmix64(seed != 0 ? seed : SeedGenerator::instance().get());
    s0_ = splitmix64.next();
    s1_ = splitmix64.next();
    s2_ = splitmix64.next();
    s3_ = splitmix64.next();
```

Reply:
Sure, fine with me.